src/components/workspace/side-content/ListVisualizer.tsx

```diff
@@ -1,24 +1,46 @@
-import { Classes } from '@blueprintjs/core';
+import { Classes, NonIdealState, Spinner } from '@blueprintjs/core';
 import * as classNames from 'classnames';
 import * as React from 'react';
 
-class ListVisualizer extends React.Component<{}, {}> {
+interface IListVisualizerState {
+  loading: boolean;
+}
+
+class ListVisualizer extends React.Component<{}, IListVisualizerState> {
   private $parent: HTMLElement | null;
 
+  constructor(props: any) {
+    super(props);
+    this.state = { loading: true };
+  }
+
   public componentDidMount() {
-    if (this.$parent) {
-      (window as any).ListVisualizer.init(this.$parent);
-    }
+    this.tryToLoad();
   }
 
   public render() {
     return (
-      <div
-        ref={r => (this.$parent = r)}
-        className={classNames('sa-list-visualizer', Classes.DARK)}
-      />
+      <div ref={r => (this.$parent = r)} className={classNames('sa-list-visualizer', Classes.DARK)}>
+        {this.state.loading && (
+          <NonIdealState description="Loading Data Visualizer..." icon={<Spinner />} />
+        )}
+      </div>
     );
   }
+
+  private tryToLoad = () => {
+    const element = (window as any).ListVisualizer;
+    if (this.$parent && element) {
+      // List Visualizer has been loaded into the DOM
+      element.init(this.$parent);
+      this.setState((state, props) => {
+        return { loading: false };
+      });
+    } else {
+      // Try again in 1 second
+      window.setTimeout(this.tryToLoad, 1000);
+    }
```

Comment:
Let's not implement `tryToLoad` or `NonIdealState` into these components. Unless we know that the init stage is failable, which I doubt so, we don't need to create "loading screens"